### DISCUSSION TITLE
hahomatic vs. homematic ip (local)

### DISCUSSION THREAD

**Discussion by SL552114** [SukramJ/aiohomematic#605] [Q&A] [answered]
Hi all,

I have been reading the documentation inside and out, but I can't really find what the difference between these two solutions is - and I apologize to the great creators of the plugin!
I am currently changing my homematic ip setup to a CCU3 and I have a Homeassistant running the OS on a RaspberryPi 4. I just installed the homematic ip (local) integration, after a bit of reading a playing around it works also with my test switch (HmIP-WRC6), which now is switching and dimming HmIP devices and non HmIP devices. So in theory all fine. But I am not sure if I am missing anything. E.g. when I restart my HA it autofinds the CCU again. Is that something I can fix somehow?
Also I am not sure if this setup auto reconnects with the CCU3, in case the CCU3 is restarted. I have seen the solution described with the system variable, but I can't find a reconnect method (maybe I am blind...).

So after all: Amazing work, thanks for all the great coding, but would appreciate if someone can help making the last bits and pieces possible as well!

Cheers,
Holger

**Comment by SukramJ** [comment, marked as answer]
**Homematic(IP) Local** is the name of the custom integration (`custom_homematic`).
`hahomematic` is the backend librarey used by `custom_homematic` for device communication.

The integration supports auto reconnect so no further action is needed. Just wait.

**Reply by SL552114** [reply]
understood, but would I need to manually install also `hahomatic`?

And is it expected behavior to see a new `custom_homematic` instance to show with the already setup CCU3 after HA startup?

**Reply by SukramJ** [reply]
> understood, but would I need to manually install also hahomatic?

No.

> And is it expected behavior to see a new custom_homematic instance to show with the already setup CCU3 after HA startup?

No. Click on ignore on the detected integration or open a new issue and fill out all requested data.
